### PR TITLE
Bv crash

### PR DESCRIPTION
**Pull request by spireken**
The build I uploaded highlights all ARCX orders with a red background to make it easier to see that the highlight works correctly. Just set a first level highlight on ARCX to show the contrast.

### CODE REVIEW

File: Applications/Spire/Source/BookView/BookQuoteTableModel.cpp [hunk 1]
```diff
@@ -103,4 +107,21 @@ void BookQuoteTableModel::on_quote_signal(const BookViewModel::Quote& quote,
     dataChanged(createIndex(index, 0), createIndex(m_size - 1,
       columnCount(QModelIndex())));
   }
+  update_market_first_indexes(index);
+}
+
+void BookQuoteTableModel::update_market_first_indexes(int index) {
```
Comment (kranar) [inline Applications/Spire/Source/BookView/BookQuoteTableModel.cpp]: The order of adjectives here is incorrect, review https://dictionary.cambridge.org/grammar/british-grammar/about-adjectives-and-adverbs/adjectives-order

File: Applications/Spire/Source/BookView/BookQuoteTableModel.cpp [hunk 2]
```diff
@@ -56,7 +56,7 @@ QVariant BookQuoteTableModel::data(const QModelIndex& index, int role) const {
     if(highlight.is_initialized()) {
       if(highlight->m_highlight_all_levels) {
         return highlight->m_color;
-      } else if(index.row() == (*m_market_first_index.find(market)).second) {
+      } else if(index.row() == (*m_first_market_index.find(market)).second) {
```
Comment (kranar) [inline Applications/Spire/Source/BookView/BookQuoteTableModel.cpp]: Use ```->``` instead of ```*```

Comment (kranar) [reply Applications/Spire/Source/BookView/BookQuoteTableModel.cpp]: Use ```[]```

File: Applications/Spire/Source/BookView/BookQuoteTableModel.cpp [hunk 3]
```diff
@@ -107,21 +107,21 @@ void BookQuoteTableModel::on_quote_signal(const BookViewModel::Quote& quote,
...
-      m_market_first_index[entry.m_code] = INT_MAX;
+      m_first_market_index[entry.m_code] = INT_MAX;
     }
   }
   for(auto i = book.rbegin() + index; i != book.rend(); ++i) {
     auto dist = std::distance(book.rbegin(), i);
-    if(dist < m_market_first_index[(**i).m_quote.m_market]) {
-      m_market_first_index[(**i).m_quote.m_market] = dist;
+    if(dist < m_first_market_index[(**i).m_quote.m_market]) {
+      m_first_market_index[(**i).m_quote.m_market] = dist;
```
Comment (kranar) [inline Applications/Spire/Source/BookView/BookQuoteTableModel.cpp]: Factor out the common sub-expression.

File: Applications/Spire/Source/BookView/BookQuoteTableModel.cpp (hunk 2 above)
Comment (spireken) [reply Applications/Spire/Source/BookView/BookQuoteTableModel.cpp]: I think find was being used because the subscript operator is non-const.

Comment (kranar) [reply Applications/Spire/Source/BookView/BookQuoteTableModel.cpp]: Okay that's good.

**Pull request merged**